### PR TITLE
chore: big docs overhaul

### LINKED ISSUE #432: write msi docs
link to https://volks73.github.io/cargo-wix/cargo_wix/ at very least

### PR DESCRIPTION
**Pull request by Gankra**
fixes #432

### CODE REVIEW

**Comment by Gankra** [comment]
image

**Comment by Gankra** [comment]
NOTE: this is blocked on cutting axolotlsay 0.2.0 with https://github.com/axodotdev/axolotlsay/pull/11

**Comment by Gankra** [comment]
HALF-BAKED: I started rebranding "executable-zip" as "archive" but haven't done all the changes

File: README.md [hunk 1]
```diff
@@ -8,18 +8,77 @@
...
+* Pick good build flags for "shippable binaries"
+* Make [tarballs][] and [installers][] for the resulting binaries
+* Generate [machine-readable manifests][manifest] so other tools can understand the results
+
+That's a short list because "we make [installers][]" is doing a lot of heavy lifting. Each installer could be (and sometimes is!) an entire standalone tool with its own documentation and ecosystem.
+
+
+## Distributing
+
+As a distribution tool, cargo-dist gets to flex its biggest superpower: **it generates its own CI scripts**. All you need to do is enable the GitHub CI backend, and `cargo dist init` will generate release.yml, which implements the full pipeline of plan, build, host, publish, announce:
```
Comment (mistydemeo) [inline README.md:52]: Maybe add a footnote indicating we'll add other backends soon?

File: README.md [hunk 2]
```diff
@@ -8,18 +8,77 @@
...
+    * Builds your [binaries and tarballs][tarballs]
+    * Builds [installers][] for your binaries
+* Publish:
+    * Uploads to package managers
+* Host + Announce:
+    * Creates (or edits) a GitHub Release
+    * Uploads build artifacts to the Release
+    * Adds relevant release notes from your RELEASES/CHANGELOG
+
+(Ideally "host" would come cleanly before "publish", but GitHub Releases doesn't really properly support this kind of staging, so we're forced to race the steps a bit here. Future work may provide a more robust release process.)
```
Comment (mistydemeo) [inline README.md:69]: I think this is too much detail to introduce at this stage of the README. People are still just reading about what we do.

File: README.md [hunk 3]
```diff
@@ -8,18 +8,77 @@
...
+
+Most of the scripts roughly amount to "install cargo-dist", "run it exactly once", "upload the artifacts it reported". **We want you to be able to copy that one cargo-dist invocation CI did, run it on your machine, and get the same results without any fuss** (not to bit-level precision, but to the kinds of precision normal people expect from cargo builds). No setting up docker, no weird linux-only shell scripts that assume a bunch of tools were setup in earlier CI steps.
+
+Of course even if we perfectly achieve this ideal, "you *can* run it locally" and "you *want to* run it locally" are different statements.
+
+To that point, **release.yml can now run partially in pull-requests**. The default is to only run the "plan" step, which includes many integrity checks to help prevent "oops the release process is broken and we only found out when we tried to cut a release".
```
Comment (mistydemeo) [inline README.md:75]: This is the README, so it's the first thing new users are seeing; I wonder if this is also too much detail, or at least expects the reader to know too much about how we used to do this?

File: book/src/artifacts/archives.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
+# Archives
+
+Archives are the primary output of cargo-dist: a zip (or tarball) containing prebuilt executables/binaries for an app, along with additional static files like READMEs, LICENSES, and CHANGELOGs. The docs previously referred to these as "executable-zips", so if you ever see that term floating around, this is what's being talked about.
```
Comment (mistydemeo) [inline book/src/artifacts/archives.md:3]: 👍 "executable-zip" is perfectly fine as an implementation name but I don't think it was the best thing to advertise to users. I think this is a nicer name to advertise.

File: book/src/artifacts/archives.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+## Build Flags
+
+We currently [always build with `--profile=dist`][dist-profile]
+
+By default we build with `--workspace` [to keep things consistent][workspace-hacks], but this can be configured with the [precise-builds config][config-precise-builds] (see those docs for details on when precise-builds will be force-enabled).
+
+By default we build your packages with default features, but this can be configured with the [features][config-features], [default-features][config-default-features], and [all-features][config-all-features] configs.
+
+When targeting windows-msvc we will unconditionally [append "-Ctarget-feature=+crt-static"][crt-static] to your RUSTFLAGS, which should just be the default for rustc but isn't for legacy reasons.
```
Comment (mistydemeo) [inline book/src/artifacts/archives.md:56]: A link as to why we think it should be the default would be great here. As a non-Windows dev. I'd love to know more about that.

File: book/src/artifacts/checksums.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
+# Checksums
+
+By default cargo-dist will generate a matching checksum file for each artifact it generates. The default checksum is sha256, so for instance `my-app-x86_64-pc-windows-msvc.zip` will also come with `my-app-x86_64-pc-windows-msvc.zip.sha256` that tools like `sha256sum` can use. This can be configured with [the checksum config][config-checksum].
+
+[Fetching installers][fetching-installers] can also use these checksums (or ones baked into them) to validate the integrity of the files they download. With https and unsigned checksums the security benefit is minimal, but it can catch more boring problems like data corruption. Updating all fetching installers to use these checksums is [still a work in progress][issue-checksum-backlog].
```
Comment (mistydemeo) [inline book/src/artifacts/checksums.md:5]: This leads a little weirdly into the next section - it implies checking them isn't done, but then the next paragraph says it is for Homebrew. Maybe worth being explicit about that.

**Review by mistydemeo — COMMENTED** [review]
Generally, I'm feeling happy about this - I think it's a much nicer organization than we had before. I do feel like we need to make the README and the intro section of the book more distinct from each other than they are in this version. The README is introducing details that are irrelevant to a lot of people landing on our landing page or the repo, but which make sense to people reading the book.

File: book/src/artifacts/archives.md (hunk 5 above)
Comment (Gankra) [reply book/src/artifacts/archives.md:56]: The link on this line is in fact the explanation :)

**Review by mistydemeo — APPROVED** [review]

**Review by shadows-withal — APPROVED** [review]
amazing work!!!

**Pull request merged**